src/ast/CMakeLists.txt

```
@@ -24,7 +24,7 @@ add_library(ast STATIC
 )
 
 target_compile_definitions(ast PRIVATE ${BPFTRACE_FLAGS})
-target_link_libraries(ast ast_defs arch parser)
+target_link_libraries(ast PRIVATE ast_defs arch parser)
```

Let's keep consistent with the rest of the changes.
```suggestion
target_link_libraries(ast PUBLIC ast_defs arch parser)
```